sdk/keyvault/azure-security-keyvault-administration/src/settings_client.cpp

```diff
@@ -0,0 +1,154 @@
+    std::string const& settingName,
+    UpdateSettingOptions const& options,
+    const Azure::Core::Context& context) const
+{
+  std::string jsonBody;
+  {
+    auto jsonRoot = Azure::Core::Json::_internal::json::object();
+    jsonRoot[ValueField] = options.Value;
+    jsonBody = jsonRoot.dump();
+  }
+  Azure::Core::IO::MemoryBodyStream requestBody(
+      reinterpret_cast<const uint8_t*>(jsonBody.data()), jsonBody.length());
+
+  auto request = CreateRequest(HttpMethod::Patch, {SettingPathName, settingName}, &requestBody);
+  auto pRawResponse = m_pipeline->Send(request, context);
+  auto httpStatusCode = pRawResponse->GetStatusCode();
+  if (httpStatusCode != Azure::Core::Http::HttpStatusCode::Ok)
+  {
+    throw Azure::Core::RequestFailedException(pRawResponse);
+  }
+  Setting response = ParseSetting(pRawResponse->GetBody());
+  return Azure::Response<Setting>(std::move(response), std::move(pRawResponse));
+}
+
+Azure::Response<Setting> SettingsClient::GetSetting(
+    std::string const& settingName,
+    const Azure::Core::Context& context) const
+{
+  auto request = CreateRequest(HttpMethod::Get, {SettingPathName, settingName});
+  auto pRawResponse = m_pipeline->Send(request, context);
+  auto httpStatusCode = pRawResponse->GetStatusCode();
+  if (httpStatusCode != Azure::Core::Http::HttpStatusCode::Ok)
+  {
+    throw Azure::Core::RequestFailedException(pRawResponse);
+  }
+  Setting response = ParseSetting(pRawResponse->GetBody());
+  return Azure::Response<Setting>(std::move(response), std::move(pRawResponse));
+}
+
+Azure::Response<SettingsListResult> SettingsClient::GetSettings(
+    const Azure::Core::Context& context) const
+{
+  auto request = CreateRequest(Azure::Core::Http::HttpMethod::Get, {SettingPathName});
+  auto pRawResponse = m_pipeline->Send(request, context);
+
+  auto httpStatusCode = pRawResponse->GetStatusCode();
+  if (httpStatusCode != Azure::Core::Http::HttpStatusCode::Ok)
+  {
+    throw Azure::Core::RequestFailedException(pRawResponse);
+  }
+  SettingsListResult response;
+  {
+    const auto& responseBody = pRawResponse->GetBody();
+    auto jsonRoot = json::parse(responseBody);
+    auto settingsArray = jsonRoot[SettingNodeName];
+    for (const auto& setting : settingsArray)
+    {
+      auto const settingString = setting.dump();
+      Setting parsedSetting
+          = ParseSetting(std::vector<uint8_t>(settingString.begin(), settingString.end()));
+      response.Value.emplace_back(std::move(parsedSetting));
+    }
+  }
+  return Azure::Response<SettingsListResult>(std::move(response), std::move(pRawResponse));
+}
+
+Setting SettingsClient::ParseSetting(std::vector<uint8_t> const& responseBody) const
+{
+  Setting response;
+  {
+    auto jsonRoot
+        = Azure::Core::Json::_internal::json::parse(responseBody.begin(), responseBody.end());
+    response.Name = jsonRoot[NameField].get<std::string>();
+    response.Value = jsonRoot[ValueField].get<std::string>();
```

Comment:
Do we get `null` or `""` or an `access violation` if the field isn't present?

Reply:
you cannot have a field without name and value in json